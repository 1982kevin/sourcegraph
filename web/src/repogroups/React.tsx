import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
