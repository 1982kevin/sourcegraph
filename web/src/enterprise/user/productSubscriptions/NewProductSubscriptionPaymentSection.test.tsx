import React from 'react'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { NewProductSubscriptionPaymentSection } from './NewProductSubscriptionPaymentSection'
import { of } from 'rxjs'
import { mount } from 'enzyme'

jest.mock('./ProductSubscriptionBeforeAfterInvoiceItem', () => ({
    ProductSubscriptionBeforeAfterInvoiceItem: 'ProductSubscriptionBeforeAfterInvoiceItem',
}))

describe('NewProductSubscriptionPaymentSection', () => {
    test('new subscription', () => {
        expect(
            mount(
                <NewProductSubscriptionPaymentSection
                    accountID={null}
                    subscriptionID={null}
                    productSubscription={{ billingPlanID: 'p', userCount: 123 }}
                    onValidityChange={() => undefined}
                    _queryPreviewProductSubscriptionInvoice={() =>
                        of<GQL.IProductSubscriptionPreviewInvoice>({
                            __typename: 'ProductSubscriptionPreviewInvoice',
                            beforeInvoiceItem: null,
                            afterInvoiceItem: {
                                __typename: 'ProductSubscriptionInvoiceItem',
                                expiresAt: '2020-01-01',
                                userCount: 123,
                                plan: {} as GQL.IProductPlan,
                            },
                            prorationDate: null,
                            isDowngradeRequiringManualIntervention: false,
                            price: 12345,
                        })
                    }
                />
            )
        ).toMatchSnapshot()
    })

    test('valid change to existing subscription', () => {
        expect(
            mount(
                <NewProductSubscriptionPaymentSection
                    accountID="a"
                    subscriptionID="s"
                    productSubscription={{ billingPlanID: 'p', userCount: 123 }}
                    onValidityChange={() => undefined}
                    _queryPreviewProductSubscriptionInvoice={() =>
                        of<GQL.IProductSubscriptionPreviewInvoice>({
                            __typename: 'ProductSubscriptionPreviewInvoice',
                            beforeInvoiceItem: {
                                __typename: 'ProductSubscriptionInvoiceItem',
                                expiresAt: '2020-01-01',
                                userCount: 123,
                                plan: {} as GQL.IProductPlan,
                            },
                            afterInvoiceItem: {
                                __typename: 'ProductSubscriptionInvoiceItem',
                                expiresAt: '2021-01-01',
                                userCount: 456,
                                plan: {} as GQL.IProductPlan,
                            },
                            prorationDate: null,
                            isDowngradeRequiringManualIntervention: false,
                            price: 23456,
                        })
                    }
                />
            )
        ).toMatchSnapshot()
    })

    test('no change to existing subscription', () => {
        expect(
            mount(
                <NewProductSubscriptionPaymentSection
                    accountID="a"
                    subscriptionID="s"
                    productSubscription={{ billingPlanID: 'p', userCount: 123 }}
                    onValidityChange={() => undefined}
                    _queryPreviewProductSubscriptionInvoice={() =>
                        of<GQL.IProductSubscriptionPreviewInvoice>({
                            __typename: 'ProductSubscriptionPreviewInvoice',
                            beforeInvoiceItem: {
                                __typename: 'ProductSubscriptionInvoiceItem',
                                expiresAt: '2020-01-01',
                                userCount: 123,
                                plan: {} as GQL.IProductPlan,
                            },
                            afterInvoiceItem: {
                                __typename: 'ProductSubscriptionInvoiceItem',
                                expiresAt: '2020-01-01',
                                userCount: 123,
                                plan: {} as GQL.IProductPlan,
                            },
                            prorationDate: null,
                            isDowngradeRequiringManualIntervention: false,
                            price: 0,
                        })
                    }
                />
            )
        ).toMatchSnapshot()
    })

    test('downgrade to existing subscription', () => {
        expect(
            mount(
                <NewProductSubscriptionPaymentSection
                    accountID="a"
                    subscriptionID="s"
                    productSubscription={{ billingPlanID: 'p', userCount: 123 }}
                    onValidityChange={() => undefined}
                    _queryPreviewProductSubscriptionInvoice={() =>
                        of<GQL.IProductSubscriptionPreviewInvoice>({
                            __typename: 'ProductSubscriptionPreviewInvoice',
                            beforeInvoiceItem: {
                                __typename: 'ProductSubscriptionInvoiceItem',
                                expiresAt: '2020-01-01',
                                userCount: 123,
                                plan: {} as GQL.IProductPlan,
                            },
                            afterInvoiceItem: {
                                __typename: 'ProductSubscriptionInvoiceItem',
                                expiresAt: '2021-01-01',
                                userCount: 456,
                                plan: {} as GQL.IProductPlan,
                            },
                            prorationDate: null,
                            isDowngradeRequiringManualIntervention: true,
                            price: 23456,
                        })
                    }
                />
            )
        ).toMatchSnapshot()
    })
})
