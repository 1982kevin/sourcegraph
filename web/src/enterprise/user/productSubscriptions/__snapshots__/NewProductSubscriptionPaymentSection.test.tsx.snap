// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProductSubscriptionPaymentSection downgrade to existing subscription 1`] = `
<NewProductSubscriptionPaymentSection
  _queryPreviewProductSubscriptionInvoice={[Function]}
  accountID="a"
  onValidityChange={[Function]}
  productSubscription={
    Object {
      "billingPlanID": "p",
      "userCount": 123,
    }
  }
  subscriptionID="s"
>
  <div
    className="new-product-subscription-payment-section"
  >
    <div
      className="form-text mb-2"
    >
      <ProductSubscriptionBeforeAfterInvoiceItem
        afterInvoiceItem={
          Object {
            "__typename": "ProductSubscriptionInvoiceItem",
            "expiresAt": "2021-01-01",
            "plan": Object {},
            "userCount": 456,
          }
        }
        beforeInvoiceItem={
          Object {
            "__typename": "ProductSubscriptionInvoiceItem",
            "expiresAt": "2020-01-01",
            "plan": Object {},
            "userCount": 123,
          }
        }
        className="mb-2"
      />
      <div
        className="alert alert-danger mb-2"
      >
        Self-service downgrades are not yet supported.
         
        <a
          href="mailto:sales@sourcegraph.com?subject=Downgrade%20subscription%20s"
        >
          Contact sales
        </a>
         
        for help.
      </div>
    </div>
  </div>
</NewProductSubscriptionPaymentSection>
`;

exports[`NewProductSubscriptionPaymentSection new subscription 1`] = `
<NewProductSubscriptionPaymentSection
  _queryPreviewProductSubscriptionInvoice={[Function]}
  accountID={null}
  onValidityChange={[Function]}
  productSubscription={
    Object {
      "billingPlanID": "p",
      "userCount": 123,
    }
  }
  subscriptionID={null}
>
  <div
    className="new-product-subscription-payment-section"
  >
    <div
      className="form-text mb-2"
    >
      Total: $
      123.45
       for
       
      14 years
       (
      123 users
      )
      <LoadingSpinner
        className="icon-inline invisible"
      >
        <div
          className="loading-spinner icon-inline invisible"
        />
      </LoadingSpinner>
    </div>
  </div>
</NewProductSubscriptionPaymentSection>
`;

exports[`NewProductSubscriptionPaymentSection no change to existing subscription 1`] = `
<NewProductSubscriptionPaymentSection
  _queryPreviewProductSubscriptionInvoice={[Function]}
  accountID="a"
  onValidityChange={[Function]}
  productSubscription={
    Object {
      "billingPlanID": "p",
      "userCount": 123,
    }
  }
  subscriptionID="s"
>
  <div
    className="new-product-subscription-payment-section"
  >
    <div
      className="form-text mb-2"
    >
      <ProductSubscriptionBeforeAfterInvoiceItem
        afterInvoiceItem={
          Object {
            "__typename": "ProductSubscriptionInvoiceItem",
            "expiresAt": "2020-01-01",
            "plan": Object {},
            "userCount": 123,
          }
        }
        beforeInvoiceItem={
          Object {
            "__typename": "ProductSubscriptionInvoiceItem",
            "expiresAt": "2020-01-01",
            "plan": Object {},
            "userCount": 123,
          }
        }
        className="mb-2"
      />
    </div>
  </div>
</NewProductSubscriptionPaymentSection>
`;

exports[`NewProductSubscriptionPaymentSection valid change to existing subscription 1`] = `
<NewProductSubscriptionPaymentSection
  _queryPreviewProductSubscriptionInvoice={[Function]}
  accountID="a"
  onValidityChange={[Function]}
  productSubscription={
    Object {
      "billingPlanID": "p",
      "userCount": 123,
    }
  }
  subscriptionID="s"
>
  <div
    className="new-product-subscription-payment-section"
  >
    <div
      className="form-text mb-2"
    >
      <ProductSubscriptionBeforeAfterInvoiceItem
        afterInvoiceItem={
          Object {
            "__typename": "ProductSubscriptionInvoiceItem",
            "expiresAt": "2021-01-01",
            "plan": Object {},
            "userCount": 456,
          }
        }
        beforeInvoiceItem={
          Object {
            "__typename": "ProductSubscriptionInvoiceItem",
            "expiresAt": "2020-01-01",
            "plan": Object {},
            "userCount": 123,
          }
        }
        className="mb-2"
      />
      <div
        className="mb-2"
      >
        Amount due: $
        234.56
      </div>
    </div>
  </div>
</NewProductSubscriptionPaymentSection>
`;
